CBinding/Compiler/ICompiler.cs

```diff
@@ -30,47 +30,431 @@
+			var streamWriter = new StreamWriter (stream);
+			FilePath path = file.ParentDirectory.Combine (workingDir);
+			ProcessWrapper p = Runtime.ProcessService.StartProcess (command, args, path, monitor.Log, streamWriter, null);
+			p.WaitForExit ();
+			streamWriter.Flush ();
+			stream.Position = 0;
+			return stream;
+		}
+
+		protected BuildResult ParseGenerationResult (Stream result, ProgressMonitor monitor)
+		{
+			var results = new BuildResult ();
+			result.Position = 0;
+			var sr = new StreamReader (result);
+			var sb = new StringBuilder ();
+			string line;
+			string fileName = "";
+			int lineNumber = 0;
+			bool isWarning = false;
+
+			while ((line = sr.ReadLine ()) != null) {
+				//e.g.	CMake Warning in/at CMakeLists.txt:10 (COMMAND):
+				//or:	CMake Warning:
+				if (line.StartsWith ("CMake Warning", StringComparison.OrdinalIgnoreCase)) {
+					//reset everything and add last error or warning.
+					if (sb.Length > 0) {
+						if (isWarning)
+							results.AddWarning (BaseDirectory.Combine (fileName), lineNumber, 0, "", sb.ToString ());
+						else
+							results.AddError (BaseDirectory.Combine (fileName), lineNumber, 0, "", sb.ToString ());
+					}
+
+					sb.Clear ();
+					fileName = "";
+					lineNumber = 0;
+					isWarning = true;
+
+					// in/at CMakeLists.txt:10 (COMMAND):
+					if (line.Contains (" in ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " in ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else if (line.Contains (" at ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " at ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else {
+						string [] warning = line.Split (':');
+						if (!string.IsNullOrEmpty (warning.ElementAtOrDefault (1))) {
+							sb.Append (warning [1]);
+						}
+					}
+				} else if (line.StartsWith ("CMake Error", StringComparison.OrdinalIgnoreCase)) {
+					//reset everything and add last error or warning.
+					if (sb.Length > 0) {
+						if (isWarning)
+							results.AddWarning (BaseDirectory.Combine (fileName), lineNumber, 0, "", sb.ToString ());
+						else
+							results.AddError (BaseDirectory.Combine (fileName), lineNumber, 0, "", sb.ToString ());
+					}      
+
+					sb.Clear ();
+					fileName = "";
+					lineNumber = 0;
+					isWarning = false;
+
+					// in/at CMakeLists.txt:10 (COMMAND):
+					if (line.Contains (" in ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " in ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else if (line.Contains (" at ")) {
+						Tuple<int, string> t = GetFileAndLine (line, " at ");
+						lineNumber = t.Item1;
+						fileName = t.Item2;
+					} else {
+						string [] error = line.Split (':');
+						if (!string.IsNullOrEmpty (error.ElementAtOrDefault (1))) {
+							sb.Append (error [1]);
+						}
+					}
+				} else {
+					sb.Append (line);
+				}
+			}
+
+			return results;
+		}
+
+		public static CMakeToolchain GetToolchain (string toolchainName )
```

Comment:
This should use an extension point instead of hardcoding the names of all the toolchain.

You could simplify this method down to:

```
AddinManager.GetExtensionObjects<CMakeToolchain> ("/CBinding/Toolchains")
    .FirstOrDefault(toolchain => toolchain.Name == toolchainName );
```

Which would be much less error prone.